include/GafferSceneUI/Private/Inspector.h

```diff
@@ -343,6 +351,13 @@ class GAFFERSCENEUI_API Inspector::Result : public IECore::RefCounted
 		/// by `acquireEdit()`. This should be displayed to the user.
 		std::string editWarning() const;
 
+		/// Returns `true` if `disableEdit()` will disable the edit
+		/// at `source()`, and `false` otherwise.
+		bool canDisableEdit() const;
+		/// Disables the edit at `source()`. Throws if
+		/// `!canDisableEdit()`
+		void disableEdit() const;
```

Comment:
We seem to be lacking the equivalent of `nonEditableReason()` here - the only way to find out why you can't disable an edit is to call `disableEdit()` and catch the exception.

Reply:
Ah yes, that would be useful and is necessary for the overall messaging improvements. Added in 6e14602.

After writing it, `nonDisablableReason()` still feels a bit clumsy to me, but fits with `nonEditableReason()`...

Reply:
Since it's an imaginary word, I suppose we can spell it how we want, but should it be `nonDisableableReason()` (extra e)? No matter how we spell it, I suspect I will always read it as "nonBlaBlaReason" :(

My only other thought was to pass a `string *reasonNot` to `canDisableEdit()`, but that gets awkward with the Python bindings. I think we can live with the awkward name...

Reply:
Changed to `nonDisableableReason()` and squashed.